ophyd/device.py

```diff
@@ -800,57 +774,68 @@ def __init__(self, prefix='', *, name, kind=None, read_attrs=None,
 
     @classmethod
     def _initialize_device(cls):
+        '''Initializes the Device and all of its Components
+
+        Initializes the following attributes from the Components::
+            - _sig_attrs - dict of attribute name to Component
+            - component_names - a list of attribute names used for components
+            - _device_tuple - An auto-generated namedtuple based on all
+              existing Components in the Device
+            - _sub_devices - a list of attributes which hold a Device
```

Comment:
Seems reasonable that we'd expose it via a property, at least, so that the user isn't modifying it inadvertently. 

I didn't expose it in this PR since that'd be a rather visible API addition - but it should be up for discussion, I think.

Reply:
Agreed with all of the above.